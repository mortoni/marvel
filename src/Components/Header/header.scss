.header {
    .box {
        width: 50px;
    }
}
