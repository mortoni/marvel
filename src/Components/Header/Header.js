import React from 'react';
import { Debounce } from 'react-throttle'
import PropTypes from 'prop-types';
import { HashLoader } from 'react-spinners';
import './header.scss';

const Header = (props) => {
    const { handleInputChange, isSearching } = props;

    const boxContent = () => {
        return isSearching ? (
            <HashLoader
                color={'#dc3545'}
                loading={ isSearching }
                sizeUnit={"px"}
                size={25}
            />
        ) : '@';
    }

    return (
        <div className="jumbotron jumbotron-fluid header">
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-10 col-sm-8 col-md-6">
                        <div className="input-group">
                            <div className="input-group-prepend">
                                <span
                                    className="input-group-text box"
                                    id="basic-addon1">
                                    { boxContent() }
                                </span>
                            </div>
                            <Debounce time="400" handler="onChange">
                                <input
                                    type="text"
                                    className="form-control"
                                    aria-describedby="search"
                                    placeholder="Search..."
                                    onChange={(e) => handleInputChange(e.target.value)}/>
                            </Debounce>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}

Header.propTypes = {
    handleInputChange: PropTypes.func.isRequired,
    isSearching: PropTypes.bool,
};

export default Header
