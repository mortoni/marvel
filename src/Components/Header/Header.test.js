import React from 'react';
import Header from './Header';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import Enzyme, { render, shallow } from 'enzyme';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

describe('Testing Header component', () => {
    const props = {
        handleInputChange: () => {},
        isSearching: false
    };

    it('renders correctly', () => {
        const wrapper = shallow(<Header { ...props }/>);
        expect(wrapper).toBeDefined();
    });

    it('renders as expected', () => {
        const wrapper = shallow(<Header { ...props }/>);
        expect(wrapper).toMatchSnapshot();
    });
});
