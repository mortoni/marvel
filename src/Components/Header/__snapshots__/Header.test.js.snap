// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleInputChange={[Function]}
    isSearching={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-10 col-sm-8 col-md-6"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text box"
                  id="basic-addon1"
                >
                  @
                </span>
              </div>
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  aria-describedby="search"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="text"
                />
              </Debounce>
            </div>
          </div>
        </div>
      </div>,
      "className": "jumbotron jumbotron-fluid header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row justify-content-center"
        >
          <div
            className="col-10 col-sm-8 col-md-6"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text box"
                  id="basic-addon1"
                >
                  @
                </span>
              </div>
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  aria-describedby="search"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="text"
                />
              </Debounce>
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-10 col-sm-8 col-md-6"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text box"
                  id="basic-addon1"
                >
                  @
                </span>
              </div>
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  aria-describedby="search"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="text"
                />
              </Debounce>
            </div>
          </div>,
          "className": "row justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text box"
                  id="basic-addon1"
                >
                  @
                </span>
              </div>
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  aria-describedby="search"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="text"
                />
              </Debounce>
            </div>,
            "className": "col-10 col-sm-8 col-md-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>
                </div>,
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />
                </Debounce>,
              ],
              "className": "input-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>,
                  "className": "input-group-prepend",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "@",
                    "className": "input-group-text box",
                    "id": "basic-addon1",
                  },
                  "ref": null,
                  "rendered": "@",
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />,
                  "handler": "onChange",
                  "handlers": Array [],
                  "time": "400",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-describedby": "search",
                    "className": "form-control",
                    "onChange": [Function],
                    "placeholder": "Search...",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-10 col-sm-8 col-md-6"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>
                </div>
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />
                </Debounce>
              </div>
            </div>
          </div>
        </div>,
        "className": "jumbotron jumbotron-fluid header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row justify-content-center"
          >
            <div
              className="col-10 col-sm-8 col-md-6"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>
                </div>
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />
                </Debounce>
              </div>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-10 col-sm-8 col-md-6"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>
                </div>
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />
                </Debounce>
              </div>
            </div>,
            "className": "row justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text box"
                    id="basic-addon1"
                  >
                    @
                  </span>
                </div>
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    aria-describedby="search"
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                  />
                </Debounce>
              </div>,
              "className": "col-10 col-sm-8 col-md-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text box"
                      id="basic-addon1"
                    >
                      @
                    </span>
                  </div>,
                  <Debounce
                    handler="onChange"
                    handlers={Array []}
                    time="400"
                  >
                    <input
                      aria-describedby="search"
                      className="form-control"
                      onChange={[Function]}
                      placeholder="Search..."
                      type="text"
                    />
                  </Debounce>,
                ],
                "className": "input-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span
                      className="input-group-text box"
                      id="basic-addon1"
                    >
                      @
                    </span>,
                    "className": "input-group-prepend",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "@",
                      "className": "input-group-text box",
                      "id": "basic-addon1",
                    },
                    "ref": null,
                    "rendered": "@",
                    "type": "span",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <input
                      aria-describedby="search"
                      className="form-control"
                      onChange={[Function]}
                      placeholder="Search..."
                      type="text"
                    />,
                    "handler": "onChange",
                    "handlers": Array [],
                    "time": "400",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-describedby": "search",
                      "className": "form-control",
                      "onChange": [Function],
                      "placeholder": "Search...",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
